rest/authorization-server/src/main/java/org/eclipse/sw360/rest/authserver/security/customheaderauth/Sw360CustomHeaderAuthenticationFilter.java

```diff
@@ -0,0 +1,202 @@
+ * one has been authenticated via the ClientDetailsStore (currently inMemory,
+ * later from CouchDB), the method
+ * org.springframework.security.oauth2.provider.password.ResourceOwnerPasswordTokenGranter.getOAuth2Authentication(ClientDetails,
+ * TokenRequest) is exchanging the current {@link Authentication} object and
+ * setting a new one from the credentials given via request params "username"
+ * and "password". These credentials are validated again as well by an
+ * {@link AuthenticationManager} that can contain different
+ * {@link AuthenticationProvider}s.
+ *
+ * So what we are doing for our custom header flow is to retrieve the client
+ * information from request parameters and expect an already authenticated user
+ * id in a configurable header. We then create an {@link Authentication} object
+ * for the client and set the request params for the user from the header. In
+ * this way the standard oauth workflow can take place. We just need to add a
+ * {@link Sw360CustomHeaderAuthenticationProvider} that only uses the given
+ * username from the proxy and another custom request parameter to know that the
+ * user has already been authenticated and create the correct authentication
+ * object from these information.
+ *
+ * It is important that the authenticating webserver is removing all configured
+ * headers for this workflow that might already be set by a client (as usual).
+ * These are the ones given in the configuration file.
+ */
+public class Sw360CustomHeaderAuthenticationFilter extends GenericFilterBean {
+
+    private final Logger log = Logger.getLogger(this.getClass());
+
+    /**
+     * Has to match username extraction in
+     * {@link ResourceOwnerPasswordTokenGranter#getOAuth2Authentication(ClientDetails, TokenRequest)}
+     */
+    private static final String PARAMETER_NAME_USERNAME = "username";
+
+    /**
+     * Not available as constant in {@link OAuth2Utils}...
+     */
+    private static final String PARAMETER_NAME_CLIENT_SECRET = "client_secret";
+
+    @Value("${security.customheader.headername.email:#{null}}")
+    private String customHeaderHeadernameEmail;
+
+    @Value("${security.customheader.headername.extid:#{null}}")
+    private String customHeaderHeadernameExtid;
+
+    @Value("${security.customheader.headername.intermediateauthstore:#{null}}")
+    private String customHeaderHeadernameIntermediateAuthStore;
+
+    private boolean active;
+
+    @Autowired
+    private Sw360CustomHeaderUserDetailsProvider sw360CustomHeaderUserDetailsProvider;
+
+    @PostConstruct
+    public void postSw360CustomHeaderAuthenticationFilterConstruction() {
+        if (StringUtils.isEmpty(customHeaderHeadernameEmail) || StringUtils.isEmpty(customHeaderHeadernameExtid)
+                || StringUtils.isEmpty(customHeaderHeadernameIntermediateAuthStore)) {
+            log.warn("Filter is NOT active! Some configuration is missing. Needed config keys:\n"
```

Comment:
I changed the log to info and the message. I think we don't need to be too verbose on configurations. If all config is there, activate it, if not, then do not activate it. It would introduce only more code and more potential errors (like: enable = true but not all config given).

Is that fine for you or am I missing something?